docs/usage.md

```diff
@@ -79,43 +83,58 @@ See also `cachi2 generate-env --help`.
 
 ### Inject project files
 
+While some package managers only need an environment file to be informed of the cache locations, others may need to
+create a configuration file or edit a lockfile (or some other file in your project directory).
+
+Before starting your build, call `cachi2 inject-files` to automatically make the necessary changes in your repository
+(based on data in the fetch-deps output directory). Please do not change the absolute path to the repo between the calls
+to fetch-deps and inject-files; if it's not at the same path, the inject-files command won't find it.
+
 ```shell
 cachi2 inject-files ./cachi2-output --for-output-dir /tmp/cachi2-output
 ```
 
+The `--for-output-dir` option has the same meaning as the one used when generating environment variables.
+
 *⚠ Cachi2 may overwrite existing files. Please make sure you have no un-committed changes (that you are not prepared to
 lose) when calling inject-files.*
 
-For some package managers, to use the pre-fetched dependencies you may need to create a configuration file or edit
-a lockfile (or some other file in your project directory).
+*⚠ Cachi2 may change files if required by the package manager. This means that the git status will become dirty if
+it was previously clean. If any scripting depends on the cleanliness of a git repository and you do not want to commit
+the changes, the scripting should either be changed to handle the dirty status or the changes should be temporarily
+stashed by wrapping in `git stash && <command> && git stash pop` according to the suitability of the context.*
 
-Before starting your build, call `cachi2 inject-files` to automatically make the necessary changes in your repo (based
-on data in the fetch-deps output directory). Please do not change the absolute path to the repo between the calls to
-fetch-deps and inject-files; if it's not at the same path, the inject-files command won't find it.
+### Building the Artifact with the Pre-fetched dependencies
```

Comment:
If we're adding `Building the Artifact` as a subtitle (###), I feel like the following titles should be nested inside it (turned into #### subtitles), as they are steps to achive this (or at least that's how I see it).

Here's an example of what I'm trying to say:

### Build the artifact with the prefetech dependencies

#### Write the Dockerfile (or Containerfile)

#### Build the container

Reply:
I agree -- this is also how it was proposed from the table of contents at the beginning. Thanks for the consistency!